javascript/frameworks/cap/test/queries/cqlinjection/srv/service1.js

```diff
@@ -0,0 +1,786 @@
+      const { id } = req.data;
+      cds.read("Entity1").where("ID =" + id);
+    });
+
+    this.on("send00122", async (req) => {
+      const { id } = req.data;
+      cds.read("Entity1").where(`ID =` + id);
+    });
+
+    this.on("send00123", async (req) => {
+      const { id } = req.data;
+      cds.read("Entity1").where(`ID=${id}`);
+    });
+
+    this.on("send00124", async (req) => {
+      const { id } = req.data;
+      cds.read("Entity1").where`ID=${id}`; // FP
+    });
+
+    this.on("send00131", async (req) => {
+      const { id } = req.data;
+      cds.create("Entity1").entries({id: "" + id});
+    });
+
+    this.on("send00132", async (req) => {
+      const { id } = req.data;
+      cds.create("Entity1").entries({id: `` + id});
+    });
+
+    this.on("send00133", async (req) => {
+      const { id } = req.data;
+      cds.create("Entity1").entries({id: `${id}`});
+    });
+
+    this.on("send00141", async (req) => {
+      const { id, amount } = req.data;
+      cds.update("Entity1").set("col1 = col1" + amount).where("col1 = " + id);
+    });
+
+    this.on("send00142", async (req) => {
+      const { id, amount } = req.data;
+      cds.update("Entity1").set("col1 = col1" + amount).where(`col1 = ` + id);
+    });
+
+    this.on("send00143", async (req) => {
+      const { id, amount } = req.data;
+      cds.update("Entity1").set("col1 = col1" + amount).where(`col1 = ${id}`);
+    });
+
+    this.on("send00144", async (req) => {
+      const { id, amount } = req.data;
+      cds.update("Entity1").set("col1 = col1" + amount).where`col1 = ${id}`; // FP
+    });
+
+    this.on("send00151", async (req) => {
+      const { id } = req.data;
+      cds.insert("Entity1").entries({id: "" + id});
+    });
+
+    this.on("send00152", async (req) => {
+      const { id } = req.data;
+      cds.insert("Entity1").entries({id: `` + id});
+    });
+
+    this.on("send00153", async (req) => {
+      const { id } = req.data;
+      cds.insert("Entity1").entries({id: `${id}`});
+    });
+
+    this.on("send00161", async (req) => {
+      const { id } = req.data;
+      cds.upsert("Entity1").entries({id: "" + id});
+    });
+
+    this.on("send00162", async (req) => {
+      const { id } = req.data;
+      cds.upsert("Entity1").entries({id: `` + id});
+    });
+
+    this.on("send00163", async (req) => {
+      const { id } = req.data;
+      cds.upsert("Entity1").entries({id: `${id}`});
+    });
+
+    this.on("send00171", async (req) => {
+      const { id } = req.data;
+      cds.delete("Entity1").where("ID =" + id);
+    });
+
+    this.on("send00172", async (req) => {
+      const { id } = req.data;
+      cds.delete("Entity1").where(`ID =` + id);
+    });
+
+    this.on("send00173", async (req) => {
+      const { id } = req.data;
+      cds.delete("Entity1").where(`ID = ${id}`);
+    });
+
+    this.on("send00174", async (req) => {
+      const { id } = req.data;
+      cds.delete("Entity1").where`ID = ${id}`; // FP
```

Comment:
Is this still an FP?

Reply:
Addressed in [28ae679](https://github.com/advanced-security/codeql-sap-js/pull/200/commits/28ae6792b81810d6cb965562be59b23a6a4612ed).